File: backend/app/services/staff_id_service.py
import re

from app.models.user import User


def generate_staff_shipping_id() -> str:
    users = User.query.filter(User.shipping_id.like("STAFF-%")).all()
    max_seq = 0
    for user in users:
        match = re.match(r"STAFF-(\d+)$", user.shipping_id)
        if match:
            max_seq = max(max_seq, int(match.group(1)))
    return f"STAFF-{max_seq + 1:05d}"

File: server/src/services/staffIdService.ts
import { like } from 'drizzle-orm'
import { db } from '../db/index.js'
import { users } from '../db/schema/index.js'

export async function generateStaffShippingId(): Promise<string> {
  const rows = await db
    .select({ shippingId: users.shippingId })
    .from(users)
    .where(like(users.shippingId, 'STAFF-%'))

  let maxSeq = 0
  for (const row of rows) {
    const match = /^STAFF-(\d+)$/.exec(row.shippingId)
    if (match) {
      maxSeq = Math.max(maxSeq, parseInt(match[1]!, 10))
    }
  }
  return `STAFF-${String(maxSeq + 1).padStart(5, '0')}`
}
